RFS/src/main/java/org/opensearch/migrations/bulkload/workcoordination/OpenSearchWorkCoordinator.java

```diff
@@ -528,7 +530,9 @@ private int numWorkItemsNotYetCompleteInternal(
             var totalHits = getTotalHitsFromSearchResponse(payload);
             // In the case where totalHits is 0, we need to be particularly sure that we're not missing data. The `relation`
             // for the total must be `eq` or we need to throw an error because it's not safe to rely on this data.
-            if (totalHits == 0 && !payload.path("hits").path("total").path("relation").textValue().equals("eq")) {
+            // ES 6.8 does not include the `relation` field, so we assume it's `eq`.
```

Comment:
This is a strange comment to have in the 'central' path code, could we refactor this into the version specific implementation?

Reply:
I can update the comment -- but I think the way its coded now is safer. Are you alright with the code itself?